src/App.tsx

```diff
@@ -1,14 +1,80 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Select } from './types/Select';
+
+function hasNormalizedQuery(content: string, query: string) {
+  const normalizedQuery = query.trim().toLowerCase();
+
+  return content.toLowerCase().includes(normalizedQuery);
+}
+
+function filterBySelect(todo: Todo, selectedOption: string) : boolean {
+  if (!selectedOption || selectedOption === Select.All) {
+    return true;
+  }
+
+  if (selectedOption === Select.Active) {
+    return !todo.completed;
+  }
+
+  return todo.completed;
+}
+
+function filterTodosByQuery(todos: Todo[], query: string, selectedOption: string) {
+  if (!query && !selectedOption) {
+    return todos;
+  }
+
+  return todos.filter((todo) => (
+    hasNormalizedQuery(todo.title, query)
+    && filterBySelect(todo, selectedOption)
+  ));
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [showLoader, setShowLoader] = useState(true);
+  const [selectedOption, setSelectedOption] = useState('');
+
+  useEffect(() => {
+    getTodos().then((newTodos) => {
+      setTodos(newTodos);
+      setShowLoader(false);
+    });
+  }, []);
+
+  const visibleTodos = filterTodosByQuery(todos, query, selectedOption);
+
+  const onHandleOpenModal = (todo: Todo) => {
+    setShowModal(true);
+    setSelectedTodo(todo);
+  };
+
+  const onHandleCloseModal = () => {
+    setShowModal(false);
+    setSelectedTodo(null);
+  };
```

Comment:
You can have only 1 function for this `handleToggleModal`(combine these 2 functions into 1)
Also, pay attention to naming, read more [here](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364)